### PR TITLE
Add topological sort after composite grouping pass

### PR DESCRIPTION
**Pull request by lsy323**
When the composite has multiple output, we the program order of the MLIR module may not be topological sorted after the composite grouping pass. It happens when the producers and the consumers of the composite overlap in program order. Because the composite is inserted after the first composite output, there will be composite input lies after the composite, violating the topological order

e.g.
```
%input1
%input2
%output1
%input3
%output2
```
after composite insertion
```
%input1
%input2
%output1
composite call (%input1, %input2, %input3)
%input3
%output2
```

### CODE REVIEW

File: torch_xla/csrc/runtime/stablehlo_composite_helper.cc [hunk 1]
```diff
@@ -277,6 +278,11 @@ class BuildStableHLOCompositePass : public mlir::OperationPass<mlir::ModuleOp> {
       }
     }
 
+    if (!mlir::sortTopologically(composite_op->getBlock())) {
+      composite_op->emitError()
+          << "The graph is not acyclic after the composite creation pass.";
```
Comment (chunnienc) [inline torch_xla/csrc/runtime/stablehlo_composite_helper.cc:283]: The graph is not acyclic after BuildStableHLOCompositePass

**Review by chunnienc — APPROVED** [review]
The topological looks great, thanks!

**Review by sdasgup3 — APPROVED** [review]

File: torch_xla/csrc/runtime/stablehlo_composite_helper.cc (hunk 1 above)
Comment (lsy323) [reply torch_xla/csrc/runtime/stablehlo_composite_helper.cc:283]: Yeah, just would like to catch the error if `mlir::sortTopologically` failed.

Comment (chunnienc) [reply torch_xla/csrc/runtime/stablehlo_composite_helper.cc:283]: I mean changing the message from "after the composite creation pass" to "after BuildStableHLOCompositePass" to be more precise lol.

Comment (lsy323) [reply torch_xla/csrc/runtime/stablehlo_composite_helper.cc:283]: sg, updated

**Pull request merged**